faw/pdf-observatory/common/processor.ts

```diff
@@ -143,87 +160,115 @@ export function reprocess(decisionDefinition: DslResult, reprocessPdfGroups: Pdf
+      extraFeature.featureText,
+    );
+    fileIndicesByExtraFeatureText.set(
+      extraFeature.featureText, fileIndices
+    );
+  }
+
+  // Add the new messages so that filters can use them
+  for (const [featureText, fileIndices] of fileIndicesByExtraFeatureText.entries()) {
+    filesWithMessages.push([featureText, Array.from(fileIndices, fileIndex => [fileIndex, 1])]);
   }
 
-  // Apply DSL expressions to fill in status
-  for (const [f, dec] of pdfMap.entries()) {
+  const fileIndicesByFilterName = new Map<string, Set<number>>();
+  for (const filter of decisionDefinition.filters) {
+    const fileIndices = fileIndicesMatchingFilter(
+      filter.patterns, filter.caseInsensitive, filter.name
+    );
+    fileIndicesByFilterName.set(filter.name, fileIndices);
+  }
+
+  // Populate decision objects with filter results (boolean) and outputs (string)
+  for (const [fileIndex, decision] of decisionsByFileIndex.entries()) {
     const evalExpr = (node: DslExpression): boolean => {
       if (node.type === 'else') return true;
       if (node.type === 'not') return !evalExpr(node.id1);
       if (node.type === 'and') return evalExpr(node.id1) && evalExpr(node.id2);
       if (node.type === 'or') return evalExpr(node.id1) || evalExpr(node.id2);
       if (node.type === 'id') {
-        let fileList = pdfGroups.get(node.id1);
+        let fileList = fileIndicesByFilterName.get(node.id1);
         if (fileList === undefined) {
           throw new Error(`Undefined filter ${node.id1}`);
         }
-
-        let r = fileList.has(f);
-        let not = pdfGroupIsNegative.get(node.id1);
-        if (not) r = !r;
-        return r;
+        return fileList.has(fileIndex);
       }
       throw new Error(`Unknown node type ${(node as any).type}`);
     };
 
-    for (const k of pdfGroups.keys()) {
-      dec[`filter-${k}`] = evalExpr({type: 'id', id1: k});
+    for (const k of fileIndicesByFilterName.keys()) {
+      decision[`filter-${k}`] = evalExpr({type: 'id', id1: k});
     }
 
     for (const [oname, ocases] of decisionDefinition.outputs.entries()) {
-      dec[oname] = 'unspecified';
+      decision[oname] = 'unspecified';
       for (const [value, expression] of ocases) {
         if (expression === null) continue;
 
         if (evalExpr(expression)) {
-          dec[oname] = value;
+          decision[oname] = value;
           break;
         }
       }
     }
   }
 
-  return newPdfs as unknown as PdfDecision[];
+  const extraFeaturesByFile = new Map<string, string[]>();
+  for (const [featureText, fileIndices] of fileIndicesByExtraFeatureText.entries()) {
+    for (const fileIndex of fileIndices) {
+      let extraFeatures = extraFeaturesByFile.get(reprocessPdfGroups.files[fileIndex]);
+      if (extraFeatures === undefined) {
+        extraFeatures = [];
+        extraFeaturesByFile.set(reprocessPdfGroups.files[fileIndex], extraFeatures);
+      }
+      extraFeatures.push(featureText);
+    }
+  }
+
+  return {
```

Comment:
Self note: Need to ensure change in return didn't mess up the CI evaluation of analysis sets

Reply:
I think [this line](https://github.com/GaloisInc/FAW/pull/87/files/3e3c29c4633c0807fca72ae6849e9bcfbd80edd0#diff-cb17c1ee98cdc3cb7d29663968730ed4e86c9836a428b15f98519a392ce6cf5eR64) covers it--it's the only other place `reprocess` is called. Definitely worth confirming though.